### PR TITLE
AZ-405: Python linter

### PR DESCRIPTION
**Pull request by pmikolajczyk41**

### CODE REVIEW

File: .github/workflows/unit_tests.yml [hunk 1]
```diff
@@ -66,6 +66,25 @@ jobs:
...
+
+      - name: Install pylint and pycodestyle
+        run: |
+          python -m pip install --upgrade pip
+          pip install pylint pycodestyle
+
+      - name: Run pylint and pycodestyle
+        run: |
+          find . -type f -name "*.py" | xargs pylint \
+          --disable=missing-docstring,too-many-instance-attributes,duplicate-code,import-error \
```
Comment (kostekIV) [inline .github/workflows/unit_tests.yml:82]: can we put this in `.pylintrc` file?

File: .github/workflows/unit_tests.yml [hunk 2]
```diff
@@ -66,6 +66,25 @@ jobs:
...
+          python-version: '3.8'
+
+      - name: Install pylint and pycodestyle
+        run: |
+          python -m pip install --upgrade pip
+          pip install pylint pycodestyle
+
+      - name: Run pylint and pycodestyle
+        run: |
+          find . -type f -name "*.py" | xargs pylint \
```
Comment (kostekIV) [inline .github/workflows/unit_tests.yml:81]: we probably want to run pylint per python project: for example `pylint ./local-tests` should be enough to run for all python files in the `local-tests` project (and is much simpler).

File: .github/workflows/unit_tests.yml [hunk 3]
```diff
@@ -66,6 +66,25 @@ jobs:
...
+          --disable=missing-docstring,too-many-instance-attributes,duplicate-code,import-error \
+          --good-names-rgxs=^[_a-z][_a-z0-9]?$
+
+          find . -type f -name "*.py" | xargs pycodestyle \
```
Comment (kostekIV) [inline .github/workflows/unit_tests.yml:85]: We can also make a config for pycodestyle.

Comment (pmikolajczyk41) [reply .github/workflows/unit_tests.yml]: -> .pycodestyle file

File: .github/workflows/unit_tests.yml (hunk 2 above)
Comment (pmikolajczyk41) [reply .github/workflows/unit_tests.yml]: done

File: .github/workflows/unit_tests.yml (hunk 1 above)
Comment (pmikolajczyk41) [reply .github/workflows/unit_tests.yml]: put

File: local-tests/.pylintrc [hunk 4]
```diff
@@ -0,0 +1,13 @@
+[MESSAGES CONTROL]
+
+disable=missing-docstring,
+        too-many-instance-attributes,
+        duplicate-code,
+        import-error
```
Comment (kostekIV) [inline local-tests/.pylintrc:6]: I would remove `import-error` from here and set the pythonpath if that is the problem this disable try to help with

Comment (pmikolajczyk41) [reply local-tests/.pylintrc]: done

**Review by kostekIV — APPROVED** [review]
:+1:

**Review by maciejnems — APPROVED** [review]

**Pull request merged**